src/lib/protocols/tls.c

```diff
@@ -1365,6 +1372,44 @@ int processClientServerHello(struct ndpi_detection_module_struct *ndpi_struct,
 		    }
 		  }
 		}
+	      } else if(extension_id == 65445 /* QUIC transport parameters */) {
+		u_int16_t s_offset = offset+extension_offset;
+		uint32_t final_offset;
+		int using_var_int = is_version_with_var_int_transport_params(quic_version);
+
+		if(!using_var_int) {
+		  u_int16_t seq_len = ntohs(*((u_int16_t*)&packet->payload[s_offset]));
+		  s_offset += 2;
+	          final_offset = MIN(total_len, s_offset + seq_len);
+		} else {
+	          final_offset = MIN(total_len, s_offset + extension_len);
+		}
+
+		while(s_offset < final_offset) {
+		  u_int64_t param_type, param_len;
+
+                  if(!using_var_int) {
+		    param_type = ntohs(*((u_int16_t*)&packet->payload[s_offset]));
+		    param_len = ntohs(*((u_int16_t*)&packet->payload[s_offset + 2]));
+		    s_offset += 4;
+		  } else {
+		    s_offset += quic_len(&packet->payload[s_offset], &param_type);
```

Comment:
Probably same here: What happens if the var int stored at s_offset is an 8 byte var int?